reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/rest/handler/AbstractActionHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.reportsscheduler.rest.handler;
+
+import java.io.IOException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestChannel;
+
+/** Action handler to process REST request and handle failures. */
+public abstract class AbstractActionHandler {
+  protected final NodeClient client;
```

Comment:
why protected instead of getters?

Reply:
Only allowing subclass to access this variable. I think protected is the first thing come to my mind
https://github.com/opendistro-for-elasticsearch/anomaly-detection/blob/92a586718f93a9ab934169cab0aca42c8641f4d9/src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/AbstractActionHandler.java#L31

Reply:
would protected getter method help here?

Reply:
protected getter should work, could you explain why protected getter is preferred?

Reply:
OOP principle of encapsulation.

Reply:
will make variables private final, and use protected getters.